### PR TITLE
Elasticsearch data mount is chowned after container start

### PR DESCRIPTION
**Pull request by crassirostris**
Fix https://github.com/kubernetes/kubernetes/issues/37030

@piosz

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37219)

### CODE REVIEW

**Comment by piosz** [comment]
LGTM

**Comment by piosz** [comment]
cc @xcompass @yifan-gu

**Comment by piosz** [comment]
Removed lgtm label because it needs to be tested in CoreOS env.

**Comment by Crassirostris** [comment]
@k8s-bot cvm gce e2e test this

File: cluster/addons/fluentd-elasticsearch/es-image/run.sh [hunk 1]
```diff
@@ -22,4 +22,6 @@ export MINIMUM_MASTER_NODES=${MINIMUM_MASTER_NODES:-2}
 
 /elasticsearch_logging_discovery >> /elasticsearch/config/elasticsearch.yml
 
-/elasticsearch/bin/elasticsearch
+chown -R elasticsearch:elasticsearch /data
```
Comment (Crassirostris) [inline cluster/addons/fluentd-elasticsearch/es-image/run.sh]: Elasticsearch doesn't work as root in the newer version, so we had to introduced privilege dropping, the question is at what point

In this change, `run.sh` started working with root privileges, so at this point we will have enough permissions to perform a chown, but elasticsearch will be executed as `elasticsearch` user later

Comment (eparis) [inline cluster/addons/fluentd-elasticsearch/es-image/run.sh]: Ohhhh, I see my stoopid.  You were su-ing the other way. I'm a fool. Ignore me. I'm going to delete these comments so they don't confuse a future person.

**Comment by Crassirostris** [comment]
@piosz Confirmed in https://github.com/kubernetes/kubernetes/issues/37030#issuecomment-262348957

Can proceed with this PR and after it's merged, https://github.com/kubernetes/kubernetes/pull/37283

**Comment by piosz** [comment]
Yet another bug fix

**Comment by xcompass** [comment]
Yeh! Thanks everyone for this prompt fix!

**Pull request merged**